verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -417,6 +463,26 @@ void IndexingFactsTreeExtractor::ExtractModulePort(
     if (tag == NodeEnum::kUnqualifiedId) {
       continue;
     }
+    if (tag == NodeEnum::kDataType) {
+      const SyntaxTreeLeaf* data_type = GetTypeIdentifierFromDataType(*child);
+      // If not null this is a non primitive type and should create
+      // kDataTypeReference node for it.
+      // This data_type may be some class or interface type.
+      if (data_type != nullptr) {
+        // Create a node for this data type and append its anchor.
+        IndexingFactNode data_type_node(
+            IndexingNodeData{IndexingFactType::kDataTypeReference});
+        data_type_node.Value().AppendAnchor(
+            Anchor(data_type->get(), context_.base));
+
+        // Make the current port node child of this data type, remove it and
+        // push the kDataTypeRefernceNode.
+        data_type_node.NewChild(facts_tree_context_.top().Children().back());
+        facts_tree_context_.top().Children().pop_back();
+        facts_tree_context_.top().NewChild(data_type_node);
```

Comment:
If this reshape pattern is common, maybe factor out a helper function like (help me think of a better name):

```c++
// illustrate transformation...
void PushDownLastChild(Node* parent, Node* new_child) {
  new_child->NewChild(parent->Children().back());
  parent->Children().pop_back();
  parent->NewChild(*new_child);
}
```

// TODO(me): try to improve this using move semantics, avoid a deep-copy where possible